File: src/works.ts
import axios from "axios";

const BASE_URL = "https://api.openalex.org";

export async function searchWorks(query: string) {
  const response = await axios.get(`${BASE_URL}/works`, {
    params: { search: query },
  });
  return response.data;
}

export async function getWorkById(id: string) {
  const response = await axios.get(`${BASE_URL}/works/${id}`);
  return response.data;
}

File: src/works.py
import requests

BASE_URL = "https://api.openalex.org"

def search_works(query: str):
    response = requests.get(f"{BASE_URL}/works", params={"search": query})
    return response.json()

def get_work_by_id(id: str):
    response = requests.get(f"{BASE_URL}/works/{id}")
    return response.json()